beacon-chain/core/helpers/weak_subjectivity.go

```diff
@@ -95,3 +97,47 @@ func ComputeWeakSubjectivityPeriod(st iface.ReadOnlyBeaconState) (types.Epoch, e
 
 	return types.Epoch(wsp), nil
 }
+
+// IsWithinWeakSubjectivityPeriod verifies if a given weak subjectivity checkpoint is not stale i.e.
+// the current node is so far beyond, that a given state and checkpoint are not for the latest weak
+// subjectivity point. Provided checkpoint still can be used to double-check that node's block root
+// at a given epoch matches that of the checkpoint.
+//
+// Reference implementation:
+// https://github.com/ethereum/eth2.0-specs/blob/master/specs/phase0/weak-subjectivity.md#checking-for-stale-weak-subjectivity-checkpoint
+
+// def is_within_weak_subjectivity_period(store: Store, ws_state: BeaconState, ws_checkpoint: Checkpoint) -> bool:
+//    # Clients may choose to validate the input state against the input Weak Subjectivity Checkpoint
+//    assert ws_state.latest_block_header.state_root == ws_checkpoint.root
+//    assert compute_epoch_at_slot(ws_state.slot) == ws_checkpoint.epoch
+//
+//    ws_period = compute_weak_subjectivity_period(ws_state)
+//    ws_state_epoch = compute_epoch_at_slot(ws_state.slot)
+//    current_epoch = compute_epoch_at_slot(get_current_slot(store))
+//    return current_epoch <= ws_state_epoch + ws_period
+func IsWithinWeakSubjectivityPeriod(
+	currentEpoch types.Epoch, wsState iface.ReadOnlyBeaconState, wsCheckpoint *eth.WeakSubjectivityCheckpoint) (bool, error) {
+	// Make sure that incoming objects are not nil.
+	if wsState == nil || wsState.LatestBlockHeader() == nil || wsCheckpoint == nil {
```

Comment:
I dont think `wState == nil` is a valid check here, you can provide a nil state object and its interface will not be nil. 
https://play.golang.org/p/5nfFudQ1ewa

Maybe do a check for the actual state ?

Reply:
What do you mean by checking for the actual state?

Reply:
Ah nvm, I was under the impresion we had a public `HasInnerState` method to check for the inner state, but seems like it was in fact changed to private a few months back.